### PR TITLE
Use cmake instead of qmake as build system

### PR DESCRIPTION
**Pull request by jpetersen**
Remove old build system and unused files.

Does not install tests anymore (for now). Does not build documentation
for maliit-glib (for now). Wayland support needs to be fixed also.

Fixes: maliit/framework#18

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+option(enable-qt5-inputcontext "Compile with Qt5 input context" ON)
+
+option(enable-hwkeyboard "Enable support for the hardware keyboard" ON)
+option(enable-dbus-activation "Enable dbus activation support for maliit-server" OFF)
+
+# Install paths
+
+if(NOT DEFINED INCLUDE_INSTALL_DIR)
+    set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH
+        "Installation directory for C header files [CMAKE_INSTALL_PREFIX/include]")
```
Comment (pvuorela) [inline CMakeLists.txt]: C++?

File: src/maliit-plugins.prf.in [hunk 2]
```diff
@@ -1,3 +0,0 @@
-LIBS *= -L@LIBDIR@ -l@MALIIT_PLUGINS_LIB@
```
Comment (pvuorela) [inline src/maliit-plugins.prf.in]: Removing the offered .prf files affect also the depending projects. Suggest not doing it.

File: CMakeLists.txt (hunk 1 above)
Comment (jpetersen) [inline CMakeLists.txt]: Ah yes of course

File: src/maliit-plugins.prf.in (hunk 2 above)
Comment (jpetersen) [inline src/maliit-plugins.prf.in]: Yes true it makes sense to still provide prf files. I will add them back

**Comment by jpetersen** [comment]
Updated the pull request (adding back *.prf files and some more smaller changes)

**Comment by jpetersen** [comment]
Merged

**Pull request closed without merging**